src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -780,6 +781,42 @@ public static bool IsValidFilePath([NotNullWhen(true)] string? fullPath)
         public static string NormalizeWithForwardSlash(string p)
             => DirectorySeparatorChar == '/' ? p : p.Replace(DirectorySeparatorChar, '/');
 
+        /// <summary>
+        /// Replaces all sequences of '\' or '/' with a single '/' but preserves UNC prefix '//'.
+        /// </summary>
+        public static string CollapseWithForwardSlash(string p)
+        {
+            var sb = new StringBuilder(p.Length);
+
+            int start = 0;
+            if (p.Length > 1 && IsAnyDirectorySeparator(p[0]) && IsAnyDirectorySeparator(p[1]))
+            {
+                // Preserve UNC paths.
+                sb.Append("//");
+                start = 2;
+            }
+
+            bool wasDirectorySeparator = false;
+            for (int i = start; i < p.Length; i++)
+            {
+                if (IsAnyDirectorySeparator(p[i]))
+                {
+                    if (!wasDirectorySeparator)
+                    {
+                        sb.Append('/');
+                    }
+                    wasDirectorySeparator = true;
```

Comment:
Consider an input string of `a\\\b`. That would result in an output of `a/b` Is that intended here? If so let's make sure there is a unit test for that case.

Reply:
That's intended, will add a test.